### ISSUE TITLE
Build failure without inclusion of QKeyEvent header

### ISSUE THREAD

**Issue by meanjollies** [Jmgr/actiona#105]
The following error appears during a build attempt:
```
keyboardkeyedit.cpp: In member function ‘virtual void ActionTools::KeyboardKeyEdit::keyPressEvent(QKeyEvent*)’:
keyboardkeyedit.cpp:54:18: error: invalid use of incomplete type ‘class QKeyEvent’
   54 |             event->ignore();
      |                  ^~
In file included from /usr/include/qt/QtWidgets/qframe.h:44,
                 from /usr/include/qt/QtWidgets/qlineedit.h:44,
                 from /usr/include/qt/QtWidgets/QLineEdit:1,
                 from codelineedit.h:27,
                 from keyboardkeyedit.h:24,
                 from keyboardkeyedit.cpp:21:
/usr/include/qt/QtWidgets/qwidget.h:75:7: note: forward declaration of ‘class QKeyEvent’
   75 | class QKeyEvent;
      |       ^~~~~~~~~
keyboardkeyedit.cpp:63:39: warning: ‘QList<T> QSet<T>::toList() const [with T = ActionTools::KeyboardKey]’ is deprecated: Use values() instead. [-Wdeprecated-declarations]
   63 |         m_keys = m_pressedKeys.toList();
      |                                       ^
In file included from /usr/include/qt/QtCore/QSet:1,
                 from abstractcodeeditor.h:26,
                 from codelineedit.h:25,
                 from keyboardkeyedit.h:24,
                 from keyboardkeyedit.cpp:21:
/usr/include/qt/QtCore/qset.h:251:14: note: declared here
  251 |     QList<T> toList() const { return values(); }
      |              ^~~~~~
keyboardkeyedit.cpp:67:14: error: invalid use of incomplete type ‘class QKeyEvent’
   67 |         event->accept();
      |              ^~
In file included from /usr/include/qt/QtWidgets/qframe.h:44,
                 from /usr/include/qt/QtWidgets/qlineedit.h:44,
                 from /usr/include/qt/QtWidgets/QLineEdit:1,
                 from codelineedit.h:27,
                 from keyboardkeyedit.h:24,
                 from keyboardkeyedit.cpp:21:
/usr/include/qt/QtWidgets/qwidget.h:75:7: note: forward declaration of ‘class QKeyEvent’
   75 | class QKeyEvent;
      |       ^~~~~~~~~
keyboardkeyedit.cpp: In member function ‘virtual void ActionTools::KeyboardKeyEdit::keyReleaseEvent(QKeyEvent*)’:
keyboardkeyedit.cpp:81:14: error: invalid use of incomplete type ‘class QKeyEvent’
   81 |         event->accept();
      |              ^~
In file included from /usr/include/qt/QtWidgets/qframe.h:44,
                 from /usr/include/qt/QtWidgets/qlineedit.h:44,
                 from /usr/include/qt/QtWidgets/QLineEdit:1,
                 from codelineedit.h:27,
                 from keyboardkeyedit.h:24,
                 from keyboardkeyedit.cpp:21:
/usr/include/qt/QtWidgets/qwidget.h:75:7: note: forward declaration of ‘class QKeyEvent’
   75 | class QKeyEvent;
      |       ^~~~~~~~~
make[1]: *** [Makefile:2754: generated/keyboardkeyedit.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/meanjollies/actiona/src/actiona-3.10.0/actiontools'
make: *** [Makefile:99: sub-actiontools-make_first-ordered] Error 2
```
I'm not sure when this problem was introduced, but I suspect it's related to the recent release of Qt 5.14, which I'm using. I was able to get it to succeed by adding `#include <QKeyEvent>` to _actiontools/keyboardkeyedit.cpp_, although maybe that isn't the right place.

**Comment by suy** [comment]
This kind of things happen often when we don't bother adding the headers of all the classes that we use, but the code still compiles with most versions of Qt because that specific header gets added in other class. I think your fix is most likely right.

**Comment by Jmgr** [comment]
Thanks for the report @meanjollies. This is indeed happening with recent Qt versions because they have removed some `#include` internally, so that `QKeyEvent` is no more present in the preprocessed C++ file. Your fix is the right one. I've fixed this in 0cfe2ed0d265fe610915b5447ca61f4bfaf01715.
